pkg/collector/corechecks/oracle/activity.go

```diff
@@ -0,0 +1,211 @@
+    CASE WHEN blocking_session_status = 'VALID' THEN
+	  blocking_instance
+	ELSE
+	  null
+	END blocking_instance,
+    CASE WHEN blocking_session_status = 'VALID' THEN
+		blocking_session
+	ELSE
+		null
+	END blocking_session,
+	CASE WHEN final_blocking_session_status = 'VALID' THEN
+    	final_blocking_instance
+	ELSE
+		null
+	END final_blocking_instance,
+	CASE WHEN final_blocking_session_status = 'VALID' THEN
+    	final_blocking_session
+	ELSE
+		null
+	END final_blocking_session,
+    CASE WHEN state = 'WAITING' THEN
+		event
+	ELSE
+		'CPU'
+	END event,
+	CASE WHEN state = 'WAITING' THEN
+    	wait_class
+	ELSE
+		'CPU'
+	END wait_class,
+	sql_text,
+	pdb_name
+FROM sys.dd_session
+WHERE 
+	( sql_text NOT LIKE '%DD_ACTIVITY_SAMPLING%' OR sql_text is NULL ) 
+	AND (
+		NOT (state = 'WAITING' AND wait_class = 'Idle')
+		OR state = 'WAITING' AND event = 'fbar timer' AND type = 'USER'
+	)`
+
+type RowMetadata struct {
+	Commands       []string `json:"dd_commands,omitempty"`
+	Tables         []string `json:"dd_tables,omitempty"`
+	Comments       []string `json:"dd_comments,omitempty"`
+	QueryTruncated string   `json:"query_truncated,omitempty"`
+}
+
+type OracleActivityRow struct {
+	SessionID              uint64  `db:"SID" json:"sid,omitempty"`
+	SessionSerial          uint64  `db:"SERIAL#" json:"serial,omitempty"`
+	Username               *string `db:"USERNAME" json:"username,omitempty"`
+	OsUser                 *string `db:"OSUSER" json:"os_user,omitempty"`
+	Process                *string `db:"PROCESS" json:"process,omitempty"`
+	Machine                *string `db:"MACHINE" json:"machine,omitempty"`
+	Program                *string `db:"PROGRAM" json:"program,omitempty"`
+	Type                   *string `db:"TYPE" json:"type,omitempty"`
+	SqlID                  *string `db:"SQL_ID" json:"sql_id,omitempty"`
+	ForceMatchingSignature *uint64 `db:"FORCE_MATCHING_SIGNATURE" json:"force_matching_signature,omitempty"`
+	SqlPlanHashValue       *uint64 `db:"SQL_PLAN_HASH_VALUE" json:"sql_plan_hash_value,omitempty"`
+	SqlExecStart           *string `db:"SQL_EXEC_START" json:"sql_exec_start,omitempty"`
+	Module                 *string `db:"MODULE" json:"module,omitempty"`
+	Action                 *string `db:"ACTION" json:"action,omitempty"`
+	ClientInfo             *string `db:"CLIENT_INFO" json:"client_info,omitempty"`
+	LogonTime              *string `db:"LOGON_TIME" json:"logon_time,omitempty"`
+	ClientIdentifier       *string `db:"CLIENT_IDENTIFIER" json:"client_identifier,omitempty"`
+	BlockingInstance       *uint64 `db:"BLOCKING_INSTANCE" json:"blocking_instance,omitempty"`
+	BlockingSession        *uint64 `db:"BLOCKING_SESSION" json:"blocking_session,omitempty"`
+	FinalBlockingInstance  *uint64 `db:"FINAL_BLOCKING_INSTANCE" json:"final_blocking_instance,omitempty"`
+	FinalBlockingSession   *uint64 `db:"FINAL_BLOCKING_SESSION" json:"final_blocking_session,omitempty"`
+	Event                  *string `db:"EVENT" json:"event,omitempty"`
```

Comment:
I assume `Event` is a waitevent? Would it be okay to name it `WaitEvent` (that is, just rename the field name and you can keep `event` as the json name if that's what the oracle convention is).

Reply:
Yes `Event` corresponds to `WaitEvent` in other integrations. I renamed both the field and the json field accordingly. I also renamed the following fields/JSON fields to be aligned with out standard: `sql_text` to `statement`, `wait_class` to `wait_event_group`, `username` to `user` and `machine` to `client`. Further, I added `port` because I saw other integrations emit this field. 

I think getting the naming right can save us lots of time, so if you think I should change other fields, like `pdb` or `cdb` lmk.

Reply:
🙇